src/shared/unit-file.c

```diff
@@ -349,10 +357,27 @@ int unit_file_build_name_map(
                                 log_debug("%s: normal unit file: %s", __func__, dst);
                         }
 
-                        r = hashmap_put_strdup(&ids, de->d_name, dst);
-                        if (r < 0)
-                                return log_warning_errno(r, "Failed to add entry to hashmap (%s→%s): %m",
-                                                         de->d_name, dst);
+                        if (revisit_masked) {
+                                if (!path_equal(dst, "/dev/null")) {
```

Comment:
note that we allow two different methods of masking: a symlink to /dev/null and an empty file. this should really cover both

Reply:
done